pkg/env/integersetting_test.go

```diff
@@ -70,6 +72,66 @@ func TestIntegerSetting(t *testing.T) {
 			wantPanic:    true,
 			wantValue:    1,
 		},
+		"shall respect allowList when value is outside of min max range": {
+			value:        "0",
+			defaultValue: 12,
+			minOpt:       func() int { return 10 },
+			maxOpt:       func() int { return 15 },
+			allowList:    func() []int { return []int{0} },
+			wantPanic:    false,
+			wantValue:    0,
+		},
+		"shall respect allowList when the value and the default value is on allowList": {
+			value:        "0",
+			defaultValue: 5,
+			minOpt:       func() int { return 10 },
+			maxOpt:       func() int { return 15 },
+			allowList:    func() []int { return []int{0, 5} },
+			wantPanic:    false,
+			wantValue:    0,
+		},
+		"shall panic even if value is on allowList but the default is not": {
+			value:        "0",
+			defaultValue: 5,
+			minOpt:       func() int { return 10 },
+			maxOpt:       func() int { return 15 },
+			allowList:    func() []int { return []int{0} },
+			wantPanic:    true,
+			wantValue:    0,
+		},
+		"shall return default if value is not on allowList but the default is": {
+			value:        "0",
+			defaultValue: 5,
+			minOpt:       func() int { return 10 },
+			maxOpt:       func() int { return 15 },
+			allowList:    func() []int { return []int{5} },
+			wantPanic:    false,
+			wantValue:    5,
+		},
+		"only values from allowList should be accepted - default on allowList, value outside": {
+			value:            "0",
+			defaultValue:     5,
+			allowList:        func() []int { return []int{5} },
+			disallowAllOther: true,
+			wantPanic:        false,
+			wantValue:        5,
+		},
+		"default outside of allowList should yield panic": {
+			value:            "1",
+			defaultValue:     5,
+			allowList:        func() []int { return []int{0, 1, 2} },
+			disallowAllOther: true,
+			wantPanic:        true,
+			wantValue:        1,
+		},
+		"default and the value on allowList should return the value": {
+			value:            "1",
+			defaultValue:     2,
+			allowList:        func() []int { return []int{0, 1, 2} },
+			disallowAllOther: true,
+			wantPanic:        false,
+			wantValue:        1,
+		},
```

Comment:
[nit] just for completion. Should we add a test case where min < value < max and it's also in the allow list?